StreamChatClient_v3/ChatClient.swift

```diff
@@ -73,64 +67,61 @@ public class Client<ExtraData: ExtraDataTypes> {
-        // Set up event middlewares
-        let middlewares: [EventMiddleware] = [
-            // TODO: Add more middlewares
-            EventDataProcessorMiddleware<ExtraData>(database: self.persistentContainer),
-            HealthCheckFilter()
+    lazy var webSocketClient: WebSocketClient = {
+        // Create a connection request
+        let webSocketEndpoint = webSocketConnectEndpoint(userId: self.currentUserId)
+        
+        var encoder = environment.requestEncoderBuilder(config.baseURL.webSocketBaseURL, config.apiKey)
+        encoder.connectionDetailsProviderDelegate = self
+        
+        let connectEndpoint = webSocketConnectEndpoint(userId: currentUserId)
+        
+        // TODO: Add more middlewares
+        let eventMiddlewares = [
+            EventDataProcessorMiddleware<ExtraData>(database: databaseContainer)
         ]
         
-        // Create a connection request
-        let socketPayload = WebSocketPayload<ExtraData>(user: self.currentUser, token: token)
-        let webSocketEndpoint = Endpoint<EmptyResponse>(path: "connect",
-                                                        method: .get,
-                                                        queryItems: nil,
-                                                        requiresConnectionId: false,
-                                                        body: ["json": socketPayload])
+        let wsClient = environment.webSocketClientBuilder(webSocketEndpoint,
+                                                          urlSessionConfiguration,
+                                                          encoder,
+                                                          EventDecoder<ExtraData>(),
+                                                          eventMiddlewares)
         
-        let requestEncoder = DefaultRequestEncoder(baseURL: config.baseURL.webSocketBaseURL, apiKey: config.apiKey)
+        wsClient.connectionStateDelegate = self
         
-        do {
-            let request = try requestEncoder.encodeRequest(for: webSocketEndpoint)
-            
-            let wsClient = environment.webSocketClientBuilder(request,
-                                                              urlSessionConfiguration,
-                                                              EventDecoder<ExtraData>(),
-                                                              middlewares)
-            return wsClient
-            
-        } catch {
-            fatalError("Failed to initialize WebSocketClient with error: \(error)")
-        }
+        return wsClient
     }()
     
     /// The `DatabaseContainer` instance `Client` uses to store and cache data.
-    private(set) lazy var persistentContainer: DatabaseContainer = {
+    private(set) lazy var databaseContainer: DatabaseContainer = {
         do {
             if config.isLocalStorageEnabled {
                 guard let storeURL = config.localStorageFolderURL else {
```

Comment:
I think the `storeURL` is related to the current user, or not?

Reply:
Discussed in Slack